tests/lax_scipy_test.py

```diff
@@ -237,6 +238,45 @@ def testXlog1pyShouldReturnZero(self):
   def testGradOfXlog1pyAtZero(self):
     partial_xlog1py = functools.partial(lsp_special.xlog1py, 0.)
     self.assertAllClose(api.grad(partial_xlog1py)(-1.), 0., check_dtypes=False)
+ 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name":
+       "_maxdegree={}_inputsize={}".format(l_max, num_z),
+       "l_max": l_max,
+       "num_z": num_z}
+       for l_max in [3, 4, 6, 32, 64]
+       for num_z in [2, 3, 4, 64, 128]))
+  def testGenerateAssociatedLegendreFunctionWithNonnegativeOrder(
+      self, l_max, num_z):
+
+    # Generates dummy points on which the associated Legendre functions are
+    # evaluated.
+    z = np.linspace(-0.2, 0.9, num_z)
+
+    actual_p = lsp_special.lpmn(m=l_max, n=l_max, z=z)
+
+    # The expected results are obtained from scipy.
+    expected_p = np.zeros((l_max + 1, l_max + 1, num_z))
+    for i in range(num_z):
+      expected_p[:, :, i] = osp_special.lpmn(l_max, l_max, z[i])[0]
+
+    def _apply_normalization(a):
+      """Applies normalization to the associated Legendre functions."""
+      num_m, num_l, _ = a.shape
+      a_normalized = np.zeros_like(a)
+      for m in range(num_m):
+        for l in range(num_l):
+          c0 = (2.0 * l + 1.0) * osp_special.factorial(l - m)
+          c1 = (4.0 * math.pi) * osp_special.factorial(l + m)
+          c2 = np.sqrt(c0 / c1)
+          a_normalized[m, l] = c2 * a[m, l]
+      return a_normalized
+
+    # The results from scipy are not normalized and the comparison requires
+    # normalizing the results.
+    expected_p_normalized = _apply_normalization(expected_p)
+
+    self.assertAllClose(actual_p, expected_p_normalized, rtol=7e-14, atol=4e-15)
```

Comment:
I'd recommend using `self._CheckAgainstNumpy` and `self._CompileAndCheck`, which will reduce boilerplate and ensure that the function behaves the way we want it to.